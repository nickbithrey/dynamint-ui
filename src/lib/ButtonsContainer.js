import React from 'react';
import Btn from '~/lib/Button';
import { Columnar, Block } from '~/lib/Grid';

const ButtonsContainer = ({children = [], buttons = []}) => (
	<Columnar horizontalAlign={'end'} tokens={{childrenGap: 5}}>
		{renderChildren(children)}
		{buttons.map(renderButton)}
	</Columnar>
);

const renderChildren = (children) => {
	if (Array.isArray(children)) {
		return children.map(renderChild)
	} else if (children) {
		return renderChild(children, 0);
	}
}

const renderChild = (child, i) => (
	<Block key={i}>
		{child}
	</Block>
);

const renderButton = (button, i) => {
	if (!button.condition()) {
		return;
	}
	return (
		<Block key={i}>
			<Btn {...button} />
		</Block>
	);
}

export default ButtonsContainer;