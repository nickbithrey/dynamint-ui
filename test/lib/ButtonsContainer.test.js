import React from 'react';
import { shallow } from 'enzyme';
import '@babel/polyfill';

import ButtonsContainer from '~/lib/ButtonsContainer';

describe('<ButtonsContainer /> rendering', () => {

	const props = {
		buttons: [
			{
				text: 'Btn1',
				onClick: jest.fn(),
				condition: () => true
			},
			{
				text: 'Btn2',
				onClick: jest.fn(),
				condition: () => true
			}
		]
	};
	
	let wrapper;
	
	beforeEach(() => {
		wrapper = shallow( <ButtonsContainer {...props} />);
	})
	
	it('can render standard buttons container', () => {
		expect(wrapper).toMatchSnapshot();
	});
	
	it('contains all buttons passed into props', () => {
		expect(wrapper.find('Button')).toHaveLength(props.buttons.length);
		props.buttons.forEach((b,i) => {
			Object.entries(b).forEach(([key,val]) => {
				expect(wrapper.find('Button').at(i).prop(key)).toBe(val);
			});
		});
	});
	
	it('renders children as well as button props', () => {
		const ButtonsContainerWithChildren = (
			<ButtonsContainer {...props}>
				<p>tag</p>
			</ButtonsContainer>
		);
		wrapper = shallow(ButtonsContainerWithChildren);
		expect(wrapper.find('Button')).toHaveLength(props.buttons.length);
		expect(wrapper.find('p')).toHaveLength(1);
	});
	
	it('renders children list as well as button props', () => {
		const ButtonsContainerWithChildren = (
			<ButtonsContainer {...props}>
				<p>tag</p>
				<p>tag2</p>
			</ButtonsContainer>
		);
		wrapper = shallow(ButtonsContainerWithChildren);
		expect(wrapper.find('Button')).toHaveLength(props.buttons.length);
		expect(wrapper.find('p')).toHaveLength(2);
		expect(wrapper.find('p').at(0).text()).toBe('tag');
		expect(wrapper.find('p').at(1).text()).toBe('tag2');
	});
		
});