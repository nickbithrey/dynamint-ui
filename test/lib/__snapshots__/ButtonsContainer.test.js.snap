// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonsContainer /> rendering can render standard buttons container 1`] = `
<Columnar
  horizontalAlign="end"
  tokens={
    Object {
      "childrenGap": 5,
    }
  }
>
  <Block
    key="0"
  >
    <Button
      condition={[Function]}
      onClick={[MockFunction]}
      text="Btn1"
      type="standard"
    />
  </Block>
  <Block
    key="1"
  >
    <Button
      condition={[Function]}
      onClick={[MockFunction]}
      text="Btn2"
      type="standard"
    />
  </Block>
</Columnar>
`;
